SECURITY.md

```diff
@@ -5,13 +5,16 @@
 Security fixes will be backported only to the rustls versions for which the
 original semver-compatible release was published less than 2 years ago.
 
-For example, as of 2023-06-13 the latest release is 0.21.1.
+For example, as of 2024-04-18 the latest release is 0.23.4.
 
+* 0.23.0 was released in February of 2024
+* 0.22.0 was released in December of 2023
 * 0.21.0 was released in March of 2023
 * 0.20.0 was released in September of 2021
```

Comment:
In case it comes up: the [GitHub release](https://github.com/rustls/rustls/releases/tag/v%2F0.20.0) for 0.20.0 was created Jul 7, 2023 but the [original release](https://github.com/rustls/rustls/commit/bc754a4fbb586beb7b1dfce38ab880fd90c0e422) was made Sept 26th, 2021. The newer date on the release is an artifact of from back-filling GitHub releases when we switched to using that process. 0.20.x is uncontroversially outside the 2 year support window.

Reply:
Maybe we should explicitly mention here that we count the crates.io release event, not the GitHub release event?

Reply:
Seems worthwhile :+1: I added a bit of warning text.

Reply:
I feel like the load-bearing part I was hoping for is specifically calling out the crates.io version as acting as the definitive date, which this doesn't do -- on purpose?

Reply:
I pushed a revision that mentions the crates.io publication date as the definitive source. Does this version read better to you?